lms/djangoapps/certificates/signals.py

```diff
@@ -79,6 +85,31 @@ def listen_for_passing_grade(sender, user, course_id, **kwargs):  # pylint: disa
 
     If needed, generate a certificate task.
     """
+    context = contexts.course_context_from_course_id(course_id)
+    if enterprise_enabled():
+        enterprise_uuids = EnterpriseCourseEnrollment.get_enterprise_uuids_with_user_and_course(
+            user.id,
+            str(course_id)
+        )
+        context["enterprise_uuid"] = str(enterprise_uuids[0])
+
+    data = {
+        'user_id': user.id,
+        'username': user.username,
+        'course_id': str(course_id),
+    }
+    segment_properties = {
+        'category': 'conversion',
+        'label': str(course_id),
+        'org': course_id.org,
+        'course': course_id.course,
+        'run': course_id.run,
+        'email': user.email
+    }
+    event_name = 'edx.course.completed'
+    with tracker.get_tracker().context(event_name, context):
+        tracker.emit(event_name, data)
+        segment.track(user.id, event_name, segment_properties)
```

Comment:
This looks good in general, my suggestion would be to wrap this whole block of code in a private helper method like `_emit_course_completed_tracking_event(user, course_id)` to that this function is easier to read at a glance.

Reply:
@iloveagent57 its done